### PR TITLE
Prepare v1.15 stable branch

### PR DESCRIPTION
**Pull request by joestringer**
Requires:
- https://github.com/cilium/cilium/pull/29802
- https://github.com/cilium/cilium/pull/29834

The main differences for the branch prep here compared to commit
4fae66689ad4 ("Prepare v1.14 stable branch") are:
- Ariane simplified some of the per-branch CI maintenance, so we don't
  need to delete all the workflows hosted in the main branch any more;
- Related, several of the conformance tests don't need to be deleted any
  more, as they can be triggered from the branch version of the workflow
- There's a few new workflows this release.

### CODE REVIEW

File: .github/maintainers-little-helper.yaml
Comment (aanm) [inline .github/maintainers-little-helper.yaml]: I think this was changed accidentally?

File: .github/workflows/lint-codeowners.yaml
Comment (aanm) [inline .github/workflows/lint-codeowners.yaml:1]: Why aren't we keeping this file?

File: .github/workflows/lint-workflows.yaml
Comment (aanm) [inline .github/workflows/lint-workflows.yaml:5]: I think we can remove this one. The WF has been disabled for a long time https://github.com/cilium/cilium/actions/workflows/lint-codeql.yaml

File: install/kubernetes/cilium/Chart.yaml
Comment (aanm) [inline install/kubernetes/cilium/Chart.yaml:7]: We haven't disclaimed anywhere that we aren't supporting k8s < 1.26. Maybe we should.

**Review by aanm — CHANGES_REQUESTED** [review]
Plus the other review I left before.

**Review by aanm — DISMISSED** [review]
I think `install/kubernetes/Makefile.values` also needs some branch-specific changes.

**Review by qmonnet — APPROVED** [review]
Apart from the items raised by André, all seems in order to me. Thanks!

**Review by nbusseneau — APPROVED** [review]

File: .github/workflows/lint-codeowners.yaml
Comment (joestringer) [reply .github/workflows/lint-codeowners.yaml:1]: Last time we removed it, so I removed it again this time. I didn't put much thought into it.

File: .github/maintainers-little-helper.yaml
Comment (joestringer) [reply .github/maintainers-little-helper.yaml]: I copied this from the v1.14 branch, so yes this was unintentional.

File: .github/workflows/lint-codeowners.yaml
Comment (joestringer) [reply .github/workflows/lint-codeowners.yaml:1]: I'll bring it back. If it causes problems, we can rip it back out.

File: install/kubernetes/cilium/Chart.yaml
Comment (joestringer) [reply install/kubernetes/cilium/Chart.yaml:7]: I don't think we've made some kind of announcement. It will be part of the docs page though: https://docs.cilium.io/en/latest/network/kubernetes/compatibility/#k8scompatibility .

I guess this is feedback to think about how we communicate this more clearly? Not sure whether to go and find the relevant PR and just bump its release-note to major or do something else...?

**Comment by joestringer** [comment]
> I think `install/kubernetes/Makefile.values` also needs some branch-specific changes.

Yeah, I caught this just between you starting the review and finishing the review. I fixed it up to match the similar branching commit for 1.14 stable.

File: .github/workflows/lint-codeowners.yaml
Comment (joestringer) [reply .github/workflows/lint-codeowners.yaml:1]: I'm confused, the workflow complains about I think every file, despite there being a catchall `* @cilium/tophat` match in codeowners?

https://github.com/cilium/cilium/actions/runs/7202801453/job/19621632353?pr=29838

**Comment by joestringer** [comment]
I'm not sure if the /test works with Ariane until we've merged the Ariane config as part of this PR :thinking: . We may have to merge this PR first, then in the next PR figure out whether Ariane and the required workflows are all configured correctly.

**Comment by joestringer** [comment]
Holding back the merge until we figure out the codeowners workflow. I'm somewhat inclined to just delete it since that's what we did for v1.14, but it's probably worth taking a slightly closer look to ensure it's not reporting about a real issue. I'll have to look at that later.

**Comment by joestringer** [comment]
I also looked around to see if we missed anything during the similar stable branch prep commit last time, but I couldn't find anything else relevant to inform how to further modify this preparation commit.

(I also took notes since this is changing 60+ files, this will be better documented in the release repo branching guide steps for next time).

**Comment by nbusseneau** [comment]
/test-backport-1.15

**Review by michi-covalent — APPROVED** [review]

File: .github/workflows/lint-codeowners.yaml
Comment (joestringer) [reply .github/workflows/lint-codeowners.yaml:1]: Yeah it looks like the workflow deliberately ignores the wildcard, but that's always how we've specified codeowners for the stable branches.

The other two jobs in this workflow lint are for checking if there are files referenced in codeowners that don't exist (unlikely to occur on a stale branch, and low stakes) and ensuring the codeowners that are in the file are documented (we don't add new codeowners into stable branches, only remove codeowners to simplify backport review). So I think the right answer is to delete this workflow.

**Comment by joestringer** [comment]
Only two workflows are failing:
- Codeowners lint, which doesn't make sense on stable branches (see [this thread above](https://github.com/cilium/cilium/pull/29838#discussion_r1426339583))
- Mergeability, blocked by labels on the PR.

I will push a fresh version to drop the codeowners lint workflow to resolve that failure. Otherwise assuming sufficient codeowner review, this should be good to merge.

**Comment by joestringer** [comment]
> However since the PR changed the trigger to /test-backport-1.15 this is what should be used :D

:facepalm: thanks, it's easy to miss the obvious stuff when you think too hard.

**Comment by joestringer** [comment]
I've addressed all of @aanm's comments. There are outstanding reviews from @cilium/sig-servicemesh and @cilium/sig-clustermesh , but the changes there are trivially replacing a branch name. @julianwiedmann informs me that there are a couple of main branch workflows that rely on having a v1.15 branch CI image, which doesn't exist until we merge this PR because the workflow changes to create those images exist in this PR. I think it makes sense to merge this at this point.

File: .github/workflows/build-images-hotfixes.yaml
Comment (joestringer) [inline .github/workflows/build-images-hotfixes.yaml:6]: Maybe a question for @nbusseneau and @cilium/ci-structure , could we just add vX.Y matches into the main branch workflows for most of these, or do they need to exist on the stable branch?

Comment (nbusseneau) [reply .github/workflows/build-images-hotfixes.yaml:6]: I think we can do it on `main`, however in this case we will only have a single version of the workflow and it must work for all branches.

If we don't expect much drift between build workflows it would be a good idea and would reduce backporting pressure as well.

**Comment by joestringer** [comment]
This PR missed adding the workflow for backport labels updater, because that only exists on older branches: https://github.com/cilium/cilium/pull/27876

File: .github/workflows/build-images-hotfixes.yaml
Comment (joestringer) [reply .github/workflows/build-images-hotfixes.yaml:6]: That'd be awesome. I had a variant on the same problem related to https://github.com/cilium/cilium/pull/27876 (& co.), because those workflows *only* exist on stable branches. Such workflows that are only on older releases complicate the stable branch creation workflow, making it more error prone.

Comment (joestringer) [reply .github/workflows/build-images-hotfixes.yaml:6]: cc @pippolo84 for context on this question regarding PR 27876 and similar for older branches. It would be easier from a release management perspective to avoid workflows that are only on stable branches, or where the stable branch workflow significantly diverges from main. As a release manager, when I create a new stable branch, I would prefer not to have to go through and understand every different workflow that exists on main vs. the last stable branch in order to properly prepare the new stable branch.

Comment (joestringer) [reply .github/workflows/build-images-hotfixes.yaml:6]: Filed https://github.com/cilium/cilium/pull/29902 to explore this specifically for the call-backport-label-updater workflow. Assuming we figure out the right details there, we can perhaps use that example as an example of how to update these other workflows.

File: CODEOWNERS
Comment (joestringer) [inline CODEOWNERS:234]: This line was incorrectly removed. https://github.com/cilium/cilium/pull/31139 will fix it in-tree.

**Comment by joestringer** [comment]
Looks like we also need to update the trigger phrases for ariane in future versions of this PR: https://docs.cilium.io/en/v1.15/contributing/testing/ci/#triggering-platform-tests (note that `/test-backport-1.15` is missing).

**Pull request merged**
